# Central Limit Theoram

- Normal Distribution -Distribution is always normal irrespective of sample size

- Non Normal Distribution - If there is random data of population and wanted to have distribution and as per Central Limit Theoram the Sample Size is adequate (approximately >30 sample) it applies normal distribution.

According to the **Central Limit Theorem**:

#### Mean of the Sample Mean:
\$[
\mu_{\bar{x}} = \mu
\$]

> This means the **mean of the sample means** is equal to the **population mean**.

---

#### Standard Deviation of the Sample Mean (also called Standard Error):

\$[
\sigma_{\bar{x}} = \frac{\sigma}{\sqrt{n}}
\$]

> Where:  
> - \$( \sigma \$) = Population standard deviation  
> - \( n \) = Sample size  
> - \$( \sigma_{\bar{x}} \$) = Standard deviation of the sample mean

where

μ = population mean

σ = population standar deviation

μx= Sample mean

σx = Sample Standard Deviation

n= Sample Size

x̅ =Sample Mean

# Types of Distribution 

There are various types of distribution in statistics but most commonly used distribution is normal distribution and Z distribution in our course.

# Hypothesis - 

Making a guess(not wild guess) based on assumption without scintific proof or explaining the situation based on reasonable assumption.

#### Null Hypothesis -H0 -> Decision always leads to status quo,current status/assumption doesn't Change.

#### Alternate Hypothesis -Ha-> Decision leads to opposite of H0.

## Types of Test

### One Tail and Two Tail tests

![hypothesis](hypothesis.png)

## One Tail test

#### A: Left tail test

> Example - Car company says the mileage the car is > 25KMPL.

- H0- All our cars average mileage is greater than 25KMPL
- Ha- Average Mileage is less then 25 KMPL

**Left Tailed test-** The alternate hypothesis contains the "<" sign. When Ha contains "<" sign, it is called left tailed test.

#### B: Right tail test

> Example, Amazon says it's delivery time is < 24 hours.
- H0 -  Our home delivery methods reduces delivery time
- Ha -  New method will increase hypothesis contains the ">" sign.

**Right Tailed test-** The alternate hypothesis contains the ">" sign. When Ha contains ">" sign, it is called right tailed test.

## Two Tail Test

Example - Difference in car mileage when driven by different persons.

**Note:** Two-Tailed Test: The alternative hypothesis contains the "!=" sign

#### Problem 1 – Hypothesis Testing: Car Mileage

A car manufacturing company claims that the mileage of their cars is **25 KMPL**.

From past experience, the **population standard deviation** is known to be **1.5 KMPL**.

A **car rental company** wants to test the **null hypothesis** that the **average mileage is at least 25 KMPL**.

To do this, a car rental representative drove **30 cars (sample size = 30)** from the same company and recorded the mileage of each car.  
The **average mileage of the sample** turned out to be **24.2 KMPL**.

---

###  Hypothesis

- **Null Hypothesis (H₀)**: μ ≥ 25  (Here μ represents the mean of population)
- **Alternate Hypothesis (H₁)**: μ < 25

This is a **left-tailed** test.

### Given statement:

- Population mean (claimed): **μ = 25**
- Sample mean: **x̄ = 24.2**
- Population standard deviation (known from past data): **σ = 1.5**
- Sample size: **n = 30**
- Significance level (α): Typically 0.05 (you can change if needed)

> Note:  
> The **standard deviation provided is the population standard deviation** (σ).  
> But in the code, we use the **standard error** (σ / √n) when calculating the p-value or Z-score (z-distribution is exaplained below),
> so we write `scale = 1.5 / np.sqrt(30)` in the function.

In [28]:
import numpy as np
mean=25
sample_mean=24.2
std=1.5    # It's a sample standard deviation. It can converted from population std to sample std by using formula below. \$[
\sigma_{\bar{x}} = \frac{\sigma}{\sqrt{n}}
\$]
n=30
z=(sample_mean-mean)/(std/np.sqrt(n))
z

-2.921186973360889

Population to Sample std conversion formula:
 \$[
\sigma_{\bar{x}} = \frac{\sigma}{\sqrt{n}}
\$]

Now, the above z score to be checked in the z table to get the p (probability) value.

From the below table, we find that the p value for z score of -2.92 is .0018

![Negative value](neg.png)

![Positive value](pos.png)

In [46]:
### How to find the z score using statistics tools

import scipy.stats as st
import numpy as np
p_values=st.norm.cdf(sample_mean,loc=mean,scale=std/np.sqrt(n))   # cdf: cumulative density function
p_values

0.0017435024460706768

# Important points

- Standard alpha level =0.05
- If p-Value is <0.05 then we **reject** null hypothesis(Ho)
- If p values is >0.05 then we **failed to reject** the Ho
- **p value** is also called **alpha value** 

**Conclusion-** As the p value is less than alpha (0.05), we reject the null hypothesis (mileage>=25KMPL)
not given 25 mileage so rental company will not buy the cars.

#  Type 1 Error -> If Ho is True and You reject Ho

In our exapmple , if average car mileage is really 25 or >25 KMPL and rental agency rejects it. Its a type 1 Error.


#   Type 2 Error -> If Ho is False and you failed to reject Ho

If average car mileage is really <25 but rental agency buying cars (failed to reject Ho). Its type 2 Error.

# Z Distribution / Z-Statistics

Let's transform our normal distribution into a **Z-distribution** (also known as the **standard normal distribution**),  
where the **mean = 0** and **standard deviation = 1**.

---
![z-distribution](z-distribution.png)

### Z → Z-distribution formula:

\[
Z$ = \frac{(x - \text{mean})}{\text{std}}
\$]

This is the formula used to convert a value into the Z-distribution.

![z-formula](z-formula.png)

# Statistical Tools 

Not only Z distribution helps in determining if **Ho** is correct of **Ha** is. These tools helps us to again on the same line in order make dicision / in order to conclude something .I want to do some type of test.Like Hypothesis Testing.

## 1. Ttest or Student Ttest
## 2. Anova -Analysis of Variance
## 3. Chi-Square test

## 1. Ttest or Student Ttest

A t-test is a statistical tool used in hypothesis testing to determine if there's a significant difference between the means of two groups. There are two types of Ttest

- One Sample Test
- Two Sample Test

**Note:** Conduct ttest when standard deviation is not known.

In [56]:
import pandas as pd
hyp=pd.read_csv("https://raw.githubusercontent.com/training-ml/Files/main/Hypothesis%20ttest.csv")
hyp.head()

ID  Gender  Race  SEB  School  Prog  Read  Write  Math1  Math2  SST
0   70       0     4    1       1     1    57     52     41     47   57
1  121       1     4    2       1     3    68     59     53     63   61
2   86       0     4    3       1     1    44     33     54     58   31
3  141       0     4    3       1     3    63     44     47     53   56
4  172       0     4    2       1     2    47     52     57     53   61

--> A teacher wants to teach student for a competative exam, she collected sample of the student marks from different school and based on her past experiance she believes that, these type of students will score more then 60 marks in the exam. Now she wants to conduct hypothesis test on the marks.

Ho=mu>=60

Ha=mu<=60

In [64]:
# One sample test

import scipy.stats as st
xbar=hyp['Math1'].mean()
alpha=0.05
mu=60  # Mean of population
st.ttest_1samp(hyp.Math1,mu)

TtestResult(statistic=-11.102736525713398, pvalue=1.3339158110547646e-22, df=199)

ttest gives two values: 1. statistics valu and 2. probability value (pvalue). We will only concentrate on pvalue.

### Actual Value:

Scientific notation: `1.3339158110547646e-22`  (e^-22 means, there 22 ZEROS before 1, see the actual value below in 'Decimal form')

Decimal form: `0.0000000000000000000001333915811`